v2/internal/pkg/release/local_stored_collector.go

```diff
@@ -241,68 +261,82 @@ func (o *LocalStorageCollector) ReleaseImageCollector(ctx context.Context) ([]v2
 				allImages = append(allImages, graphCopySlice...)
 			}
 		}
-		releaseCopyImages, err := o.prepareD2MCopyBatch(allRelatedImages)
-		if err != nil {
-			//o.Log.Error(errMsg, err.Error())
-			return []v2alpha1.CopyImageSchema{}, err
-		}
-		allImages = append(allImages, releaseCopyImages...)
 	}
 
 	return allImages, nil
 }
 
-func (o LocalStorageCollector) prepareM2DCopyBatch(images []v2alpha1.RelatedImage) ([]v2alpha1.CopyImageSchema, error) {
+func (o LocalStorageCollector) prepareM2DCopyBatch(images []v2alpha1.RelatedImage, releaseTag string) ([]v2alpha1.CopyImageSchema, error) {
 	var result []v2alpha1.CopyImageSchema
 	for _, img := range images {
 		var src string
 		var dest string
 
 		imgSpec, err := image.ParseRef(img.Image)
 		if err != nil {
-			//o.Log.Error("%s", err.Error())
 			return nil, err
 		}
 		src = imgSpec.ReferenceWithTransport
-		if imgSpec.IsImageByDigest() {
-			tag := fmt.Sprintf("%s-%s", imgSpec.Algorithm, imgSpec.Digest)
+		pathComponents := ""
+		tag := imgSpec.Tag
+
+		if img.Type == v2alpha1.TypeOCPRelease {
+			pathComponents = releaseImagePathComponents
+			tag = imgSpec.Tag
+		} else if img.Type == v2alpha1.TypeCincinnatiGraph {
+			pathComponents = imgSpec.PathComponent
+		} else if img.Type == v2alpha1.TypeOCPReleaseContent && img.Name != "" {
+			pathComponents = releaseComponentPathComponents
+			tag = releaseTag + "-" + img.Name
+		} else if imgSpec.IsImageByDigestOnly() {
+			pathComponents = imgSpec.PathComponent
+			tag = fmt.Sprintf("%s-%s", imgSpec.Algorithm, imgSpec.Digest)
 			if len(tag) > 128 {
 				tag = tag[:127]
 			}
-			dest = dockerProtocol + strings.Join([]string{o.destinationRegistry(), imgSpec.PathComponent + ":" + tag}, "/")
-		} else {
-			dest = dockerProtocol + strings.Join([]string{o.destinationRegistry(), imgSpec.PathComponent + ":" + imgSpec.Tag}, "/")
-
 		}
+
+		dest = dockerProtocol + strings.Join([]string{o.destinationRegistry(), pathComponents + ":" + tag}, "/")
+
 		o.Log.Debug("source %s", src)
 		o.Log.Debug("destination %s", dest)
 		result = append(result, v2alpha1.CopyImageSchema{Origin: img.Image, Source: src, Destination: dest, Type: img.Type})
 	}
 	return result, nil
 }
 
-func (o LocalStorageCollector) prepareD2MCopyBatch(images []v2alpha1.RelatedImage) ([]v2alpha1.CopyImageSchema, error) {
+func (o LocalStorageCollector) prepareD2MCopyBatch(images []v2alpha1.RelatedImage, releaseTag string) ([]v2alpha1.CopyImageSchema, error) {
 	var result []v2alpha1.CopyImageSchema
 	for _, img := range images {
 		var src string
 		var dest string
 
 		imgSpec, err := image.ParseRef(img.Image)
 		if err != nil {
-			//o.Log.Error("%s", err.Error())
 			return nil, err
 		}
-		if imgSpec.IsImageByDigest() {
-			tag := fmt.Sprintf("%s-%s", imgSpec.Algorithm, imgSpec.Digest)
+		pathComponents := ""
+		tag := imgSpec.Tag
+
+		if img.Type == v2alpha1.TypeOCPRelease {
```

Comment:
@sherine-k  same here - also looks like duplication , could this be refactored ?